File: backend/user-module/src/main/java/com/example/backend/dto/response/UserPrivacyInfoResponse.java
package com.example.backend.dto.response;

public record UserPrivacyInfoResponse(
        String name,
        String nickname,
        String phoneNumber,
        String profileImageUrl,
        String email,
        boolean isSocial
) {
    public static UserPrivacyInfoResponse of(
            String name,
            String nickname,
            String phoneNumber,
            String profileImageUrl,
            String email,
            boolean isSocial
    ){
        return new UserPrivacyInfoResponse(name, nickname, phoneNumber, profileImageUrl, email, isSocial);
    }
}

File: frontend/src/features/profile/types/UserPrivacyInfoResponse.ts
export interface UserPrivacyInfoResponse{
    name: string;
    nickname: string;
    phoneNumber: string;
    profileImageUrl: string | null;
    email: string;
    isSocial: boolean;
}

// 개인정보 변경 닉네임 중복체크
export interface ProfileNicknameCheck {
    exists: boolean;
}